File: leetcode/april2024/c++/10_reveal_cards_in_increasing_order.cpp
// CODE

class Solution {
    public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        int n = deck.size(), i = 0, j = 0;
        bool flag = false;
        vector<int> res(n, 0);
        sort(deck.begin(), deck.end());
        while(j < n) {
            if(res[i] == 0) {
                if(!flag) {
                    res[i] = deck[j];
                    j++;
                }
                flag = !flag;
            }
            i = (i + 1) % n;
        }
        return res;
    }
};

File: leetcode/2024_4_april/python/10_reveal_cards_in_increasing_order.py
# CODE

class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        n = len(deck)
        i, j = 0, 0
        flag = False
        res = [0] * n
        deck.sort()

        while j < n:
            if res[i] == 0:
                if not flag:
                    res[i] = deck[j]
                    j += 1
                flag = not flag
            i = (i + 1) % n
        
        return res